File: CouponService/Models/Coupon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace CouponService.Models;

[Index("Code", Name = "UQ__Coupons__357D4CF97B02EC47", IsUnique = true)]
public partial class Coupon
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("code")]
    [StringLength(20)]
    [Unicode(false)]
    public string Code { get; set; } = null!;

    [Column("discount", TypeName = "decimal(5, 2)")]
    public decimal Discount { get; set; }

    [Column("max_uses")]
    public int? MaxUses { get; set; }

    [Column("expiry_date")]
    public DateTime ExpiryDate { get; set; }
}

File: CouponService/src/Models/Coupon.js
class Coupon {
    constructor(id, code, discountType, discountValue, maxUses, expiryDate, occasion, minPrice, created_at, updated_at) {
        this.id = id;                    
        this.code = code;                
        this.discountType = discountType; 
        this.discountValue = discountValue; 
        this.maxUses = maxUses;          
        this.expiryDate = expiryDate;     
        this.occasion = occasion;         
        this.minPrice = minPrice;
        this.created_at = created_at;
        this.updated_at = updated_at;  
    }
}

module.exports = Coupon;
